packages/datadog-instrumentations/src/confluentinc-kafka-javascript.js

```diff
@@ -206,7 +212,8 @@ function instrumentKafkaJS (kafkaJS) {
                         channels.producerStart.publish({
                           topic: payload?.topic,
                           messages: payload?.messages || [],
-                          bootstrapServers: kafka._ddBrokers
+                          bootstrapServers: kafka._ddBrokers,
+                          disableHeaderInjection: disabledHeaderWeakMap.get(producer)
```

Comment:
Suggestion:

```suggestion
                          disableHeaderInjection: disabledHeaderWeakSet.has(producer)
```

(Also in the other file)

Reply:
done.